cf_units/__init__.py

```diff
@@ -1935,7 +1934,7 @@ def date2num(self, date):
             >>> u = cf_units.Unit('hours since 1970-01-01 00:00:00',
             ...                   calendar=cf_units.CALENDAR_STANDARD)
             >>> u.date2num(datetime.datetime(1970, 1, 1, 5))
-            5.00000000372529
+            4.999999999998181
```

Comment:
I'm just not convinced this is OK.

Reply:
This is actually closer to the target of `5` in `v1.0.2.1` than previously in `v1.0.1`.

The `cftime` guys reimplemented the algorithm to calculate a `JulianDayFromDate` - their motivation was to be more accurate and more performant. Indeed, they state that it gives approximately 10 microsecond accuracy, as opposed to approximately 1 millisecond accuracy.

See the `cftime` difference between `v1.0.1` and `v1.0.2.1` - https://github.com/Unidata/cftime/compare/v1.0.1...Unidata:v1.0.2.1

Reply:
It may be closer in precision but it's still plain wrong; that number needs to read 5 if it's to have any value at all... Otherwise all cftime is doing here is being unable to represent an hour value accurately. 

This behaviour will cause all sorts of issues in downstream systems that are dependent on being able to consistently represent an hour value – imagine a lagged ensemble that needs to accurately represent the offset from the starting point in hours. With this situation the ensemble could quite practically, and irreproducibly, end up with two 4hr lagged members, no 5hr member, and a 6hr member...

Reply:
Perhaps what I want is for cf-units to accurately mathematically round such values, so that an offset of 5hrs is accurately represented by such – an integer value of 5. As such, the result to this test would be the integer value `5`. At that point I'd be happy to accept this proposed change...

Reply:
@dkillick My plan is to address this **directly** in `cftime`. 

I've recently been working on various other fixes in `cftime`, which have been merged and as a result I've just been given collaborator status on `cftime`! :smile:

My intention is to fix the problem at source, and that's not `cf-units`. I don't think that we should patch over the behaviour of `cftime` here.... that said, I don't want us to be blocked because of `cftime`.

Are you willing to accept this PR "as is" on the promise that I then identify and address the root problem in `cftime`? 

Once `cftime` is fixed we will have to again adject the doc-test to show the correct JulianDay value, but that's okay by me, as `cf-units` won't be blocked from all PRs due to failing tests all because of `cftime`. Seems like a win-win strategy to me.

What do you think?

Reply:
Just weighing in here - there *is no precise mathematical representation of this number*. The calculation will **always** be a float, and it will always have this feature.

What I think @dkillick *might* be asking for is for the test to show the useful value. If so, I completely agree with the sentiment, and a simple:

```
"{:.2f}".format(4.99999999)
```

Will solve the doctest once-and-for-all.

Reply:
> I've just been given collaborator status

Good job!

> Are you willing to accept this PR "as is"

I'd much rather we pin to a non-broken `cftime` (1.0.1?) and then unpin once a fixed and updated `cftime` has been released...

Reply:
> there is no precise mathematical representation of this number

I disagree. We have an offset of 5 hours from an epoch, which is 5 hours, which can be measured _exactly_ as an integer value of 5. We should do this and be done with it, rather than fussing with sub-second precision when we're measuring whole hours.